### PR TITLE
HMR + Build + Style

### PR DESCRIPTION
**Pull request by renchap**
Based on #7 by @mfazekas, I rewrote the `README`, unified some styling and changed the build process to avoid messing with `dist` in paths.

@mfazekas, @daninfpj, would you mind having a look before I merge?

### CODE REVIEW

**Comment by daninfpj** [comment]
@renchap: will do later today

File: README.md [hunk 1]
```diff
@@ -77,6 +77,95 @@ You can pass any of the usual arguments to render in this call: `layout`, `statu
 
 *Note: you need to have [Webpack process your code](https://github.com/rails/webpacker#binstubs) before it is available to the browser, either by manually running `./bin/webpack` or having the `./bin/webpack-watcher` process running.*
 
+### Hot Module Replacement
+
+[HMR](https://webpack.js.org/guides/hmr-react/) allows to reload / add / remove modules live in the browser without
+reloading the page. This allows any change you make to your React components to be applied as soon as you save,
+preserving their current state.
+
+First, install `react-hot-loader` :
```
Comment (daninfpj) [inline README.md]: Remove extra space before ':'?

File: .gitignore [hunk 2]
```diff
@@ -10,6 +9,9 @@ node_modules/
 /spec/reports/
 /tmp/
 
+/javascript/webpacker_react-npm-module/dist/*.js
+/javascript/webpacker_react-npm-module/dist/yarn.lock
```
Comment (daninfpj) [inline .gitignore]: Should `/javascript/webpacker_react-npm-module/yarn.lock` be ignored as well?

File: javascript/webpacker_react-npm-module/dist/package.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (daninfpj) [inline javascript/webpacker_react-npm-module/dist/package.json]: I'm not sure I understand the need to introduce this file in addition to `../package.json`?

Comment (renchap) [inline javascript/webpacker_react-npm-module/dist/package.json]: To be able to use `import * from "webpacker-react/something" we need `something.js` to be in the root of the NPM package. With Babel transpiling, output is created in `dist/`. I havent found an alternative other than running `yarn publish` from the `dist/` directory, and the `package.json` needs to be in this directory (but will only be used for the NPM package, so I stripped it down to the minimum).

I am wondering if `javascript/webpacker_react-npm-module/package.json` (used to install Babel for transpiling, and nothing else) should be moved to the root directory. I am also considering moving the NPM scripts to some Rake tasks to ease building the various parts.

File: .gitignore (hunk 2 above)
Comment (renchap) [inline .gitignore]: Good catch, in fact `yarn.lock` should never be ignored.

**Pull request merged**